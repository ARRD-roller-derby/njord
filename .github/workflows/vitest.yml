name: Vitest
on: push
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v2
      with:
        node-version: '16.x'
    - name: Install dependencies
      run: npm ci
    - name: Run Vitest tests
      run: npm run coverage
    - uses: actions/upload-artifact@v2
      if: always()
      with:
        name: vitest-report
        path: vitest-report/
        retention-days: 30