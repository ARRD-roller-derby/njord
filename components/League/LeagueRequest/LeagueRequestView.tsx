import FullscreenModale from '../../_ui/FullscreenModale/FullscreenModale'
import LeagueRequestForm from '../LeagueRequestForm/LeagueRequestForm'

interface props {
  readonly open: boolean
  readonly openModale: Function
  readonly closeModale: Function
  readonly canIRequest: boolean
}
export default function LeagueRequestView({
  open,
  openModale,
  closeModale,
  canIRequest,
}: props) {
  return (
    <>
      <button onClick={() => openModale()}>Rejoindre une league</button>
      {open && (
        <FullscreenModale setClose={() => closeModale()}>
          {canIRequest ? (
            <LeagueRequestForm close={closeModale} />
          ) : (
            <p>
              Vous ne pouvez pas envoyer plusieurs requête dans un délai aussi
              court.
            </p>
          )}
        </FullscreenModale>
      )}
    </>
  )
}
