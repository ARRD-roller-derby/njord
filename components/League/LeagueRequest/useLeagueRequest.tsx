import { useState } from 'react'

export default function useLeagueRequest(): {
  open: boolean
  openModale: Function
  closeModale: Function
} {
  const [open, setOpen] = useState<boolean>(false)

  return {
    open,
    openModale: () => setOpen(true),
    closeModale: () => setOpen(false),
  }
}
