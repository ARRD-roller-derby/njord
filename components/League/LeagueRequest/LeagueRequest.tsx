import LeagueRequestView from "./LeagueRequestView";
import useLeagueRequest from './useLeagueRequest';

interface props {
  readonly canIRequest:boolean
}

export default function LeagueRequest({canIRequest}:props){
  const props = useLeagueRequest();
  return <LeagueRequestView {...props} canIRequest={canIRequest}/>
}