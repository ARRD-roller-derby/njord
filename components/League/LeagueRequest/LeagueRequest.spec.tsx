import { afterEach, describe, expect, it } from 'vitest'
import { render, cleanup } from '@testing-library/react'
import LeagueRequest from './LeagueRequest'

describe('<LeagueRequest />', () => {
  afterEach(cleanup)

  it('Check snapshot', () => {
    const { asFragment } = render(
      <LeagueRequest canIRequest={false} />
    )
    expect(asFragment()).toMatchSnapshot()
  })

})
