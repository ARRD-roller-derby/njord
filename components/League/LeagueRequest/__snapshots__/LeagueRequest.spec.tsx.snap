// Vitest Snapshot v1

exports[`<LeagueRequest /> > Check snapshot 1`] = `
<DocumentFragment>
  <button>
    Rejoindre une league
  </button>
</DocumentFragment>
`;
